app/grandchallenge/cases/views.py

```diff
@@ -94,6 +98,17 @@ class RawImageUploadSessionViewSet(
 ):
     serializer_class = RawImageUploadSessionSerializer
     queryset = RawImageUploadSession.objects.all()
+    permission_classes = [DjangoObjectOnlyWithCustomPostPermissions]
+    filter_backends = [ObjectPermissionsFilter]
 
     def perform_create(self, serializer):
         serializer.save(creator=self.request.user)
+
+
+class RawImageFileViewSet(
+    CreateModelMixin, RetrieveModelMixin, ListModelMixin, GenericViewSet
+):
+    serializer_class = RawImageFileSerializer
+    queryset = RawImageFile.objects.all()
+    permission_classes = [DjangoObjectOnlyWithCustomPostPermissions]
+    filter_backends = [ObjectPermissionsFilter]
```

Comment:
We need to add validation to the Serializer to check extra permissions on the POST view (see reader study answers):
- [ ] Check that the user has view permissions for the related `RawImageUploadSession`
- [ ] Add setting the algorithm field and check that the user has permission to view that algorithm

Reply:
@jmsmkn Just double checking RawImageFile  should have an "algorithm" field?